lib/components/SchematicViewer.tsx

```diff
@@ -47,6 +47,23 @@ export const SchematicViewer = ({
   )
   const svgDivRef = useRef<HTMLDivElement>(null)
 
+  const [internalEditEvents, setInternalEditEvents] = useState<ManualEditEvent[]>([])
+  const circuitJsonRef = useRef<CircuitJson>(circuitJson)
+
+  const getCircuitHash = (circuitJson: CircuitJson) => {
+    return `${circuitJson?.length || 0}_${(circuitJson as any)?.editCount || 0}`
+  }
+
+  useEffect(() => {
+    const circuitHash = getCircuitHash(circuitJson)
+    const circuitHashRef = getCircuitHash(circuitJsonRef.current)
+
+    if (circuitHash !== circuitHashRef) {
+      setInternalEditEvents([])
+      circuitJsonRef.current = circuitJson
+    }
+  }, [circuitJson])
```

Comment:
fwiw this is just a hack when circuitJson is passed as a newly constructed array, generally only true in fixtures not in production code, but it's not bad to have